### PR TITLE
feat(accounts) Login with verified emails only

### PR DESCRIPTION
**Pull request by 9mido**
Adds a setting to only allow verified emails to login. Unverified emails will get a form validation error.

This is useful because it prevents unverified emails trying to login from sending spam confirmation emails over and over. And it ensures only legitimate people are able to login to your website. 

If this setting is true, the django developer could always run a cronjob to delete unverified emails if the newly signed up user loses their original confirmation email or it expires. Then the user could signup again with the same email address.

### CODE REVIEW

File: allauth/account/forms.py [hunk 1]
```diff
@@ -183,6 +183,9 @@ def clean(self):
                 login = self.cleaned_data["login"]
                 if self._is_login_email(login):
                     auth_method = app_settings.AuthenticationMethod.EMAIL
+                    email = user.email
+                    if EmailAddress.objects.filter(email=email, verified=False).exists() is True and app_settings.LOGIN_VERIFIED_EMAIL_ONLY is True:
```
Comment (matmair) [inline allauth/account/forms.py:187]: @9mido wouldn't it be more efficient to put the settings check first? Database-access-wise I mean

Comment (9mido) [reply allauth/account/forms.py:187]: You mean change the order of the if statement to be:

`if app_settings.LOGIN_VERIFIED_EMAIL_ONLY is True and EmailAddress.objects.filter(email=email, verified=False).exists() is True:` ? I don't think it really matters much.

Comment (matmair) [reply allauth/account/forms.py:187]: Access to settings does not require a db load so first checking that would reduce dB actions. But it is marginal

Comment (9mido) [reply allauth/account/forms.py:187]: Right. Can you show me the exact code correction you would like to see and how I would change it?

Comment (matmair) [reply allauth/account/forms.py:187]: if app_settings.LOGIN_VERIFIED_EMAIL_ONLY is True and EmailAddress.objects.filter(email=email, verified=False).exists() is True:

**Comment by pennersr** [comment]
You mention two improvements:

> 1) This is useful because it prevents unverified emails trying to login from sending spam confirmation emails over and over. 

But, it essentially blocks people from recovering their account in case they lost the first verification email. Also, I can create signups for a lot of email addresses that I don't own, and all those email addresses would be blocked from signing up.

That is quite a price to pay for addressing the issue of sending too many confirmation emails. 


> And it ensures only legitimate people are able to login to your website.

I don't see why this is listed as an improvement point, as this is already the case (when email verification is mandatory).

If sending too many confirmation emails is the problem, I would rather tackle just that. For example, we could just as well implement an exponential back off for sending those mails and we solve the issue without blocking users or having to resort to cron jobs.

**Pull request closed without merging**